internal/dag/gatewayapi_processor.go

```diff
@@ -104,14 +108,22 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 			// with the Gateway. If this Selector is defined, only routes matching the Selector
 			// are associated with the Gateway. An empty Selector matches all routes.
 
-			nsMatches, err := p.namespaceMatches(listener.Routes.Namespaces, route)
-			if err != nil {
-				p.Errorf("error validating namespaces against Listener.Routes.Namespaces: %s", err)
+			var err error
+			nsMatches := true
+			selMatches := true
+
+			if listener.Routes.Namespaces != nil {
+				nsMatches, err = p.namespaceMatches(*listener.Routes.Namespaces, route)
```

Comment:
would it be a little cleaner to have `namespaceMatches` and `selectorMatches` take a pointer and just return true if the passed in namespace/selector is nil?

Reply:
👍